### PR TITLE
Fake companion in 4d

### PR DESCRIPTION
**Pull request by arainot**

### CODE REVIEW

**Comment by carlgogo** [comment]
@arainot you have to check that the code is py2 and py3 compatible. Check the Travis CI error log for a hint.

**Comment by arainot** [comment]
Works now!

File: vip_hci/phot/fakecomp.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 """
 
 from __future__ import division
-from __future__ import print_function
```
Comment (carlgogo) [inline vip_hci/phot/fakecomp.py]: Why this line is gone? It's better to keep it.

File: vip_hci/phot/fakecomp.py [hunk 2]
```diff
@@ -18,16 +17,22 @@
 from ..preproc import cube_crop_frames, frame_shift, frame_crop
 from ..var import frame_center, fit_2dgaussian, get_circle
 
+def cube_shift(cube, y, x, imlib):
```
Comment (carlgogo) [inline vip_hci/phot/fakecomp.py]: Not sure we need a function for something that could take a single line. Anyway let's leave it, but put it at the end of the file, and complete de documentation

File: vip_hci/phot/fakecomp.py [hunk 3]
```diff
@@ -47,77 +52,168 @@ def inject_fcs_cube(array, psf_template, angle_list, flevel, plsc, rad_dists,
...
+            w = int(np.floor(size_fc/2.))
+            # fcomp in the center of a zeros frame
+            if (psf_template[0].shape[1]%2)==0:
+                fc_fr[i, ceny-w:ceny+w, cenx-w:cenx+w] = psf_template[i]
+            else:
+                fc_fr[i, ceny-w:ceny+w+1, cenx-w:cenx+w+1] = psf_template[i]
+
+        array_out = np.zeros_like(array)
+
+        #if len(theta)>1:
```
Comment (carlgogo) [inline vip_hci/phot/fakecomp.py]: Development and commented lines like this should be avoided, remove

File: vip_hci/phot/fakecomp.py [hunk 4]
```diff
@@ -47,77 +52,168 @@ def inject_fcs_cube(array, psf_template, angle_list, flevel, plsc, rad_dists,
...
+            else:
+                for branch in range(n_branches):
+                    print 'Branch '+str(branch+1)+':'
+                    for i in range(n_fc_rad):
+                        ang = (branch * 2 * np.pi / n_branches) + np.deg2rad(theta)
+                        posy = rad_dists[i] * np.sin(ang) + ceny
+                        posx = rad_dists[i] * np.cos(ang) + cenx
+                        rad_arcs = rad_dists[i]*plsc
+                        msg ='\t(X,Y)=({:.2f}, {:.2f}) at {:.2f} arcsec ({:.2f} pxs)'
+                        #print msg.format(posx, posy, rad_arcs, rad_dists)
```
Comment (carlgogo) [inline vip_hci/phot/fakecomp.py]: I don't get this if else structure. In the else case you are not printing anything

File: vip_hci/phot/fakecomp.py [hunk 5]
```diff
@@ -148,40 +244,41 @@ def create_psf_template(array, size, fwhm=4, verbose=True, collapse='mean'):
...
+    if verbose:
+        print "Done scaled PSF template from the average of", n,"frames."
     return psf_normd
 
 
 def psf_norm(array, fwhm=4, size=None, threshold=None, mask_core=None,
              full_output=False, verbose=False):
     """ Scales a PSF, so the 1*FWHM aperture flux equals 1.
-    
+
```
Comment (carlgogo) [inline vip_hci/phot/fakecomp.py]: @arainot Why there are changes in the psf_norm function at all? It complicates the review. The logic of this function was not supposed to be modified, yet some lines appear to be changed. Please, take care of this in the future, only touch the lines (and characters) that need to be modified.

**Comment by arainot** [comment]
@carlgogo I updated the code

**Comment by carlgogo** [comment]
@arainot thanks!

**Pull request merged**